modules/nextflow/src/main/groovy/nextflow/script/FusionMetadata.groovy

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2013-2024, Seqera Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package nextflow.script
+
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+import groovy.transform.CompileStatic
+import groovy.transform.EqualsAndHashCode
+import groovy.transform.ToString
+import groovy.util.logging.Slf4j
+import nextflow.Session
+/**
+ * Models Fusion metadata for Nextflow execution
+ * 
+ * @author Marco De La Pierre <marco.delapierre@gmail.com>
+ */
+@Slf4j
+@CompileStatic
+@ToString(includeNames = true)
+@EqualsAndHashCode
+class FusionMetadata {
+    final private Pattern VERSION_JSON = Pattern.compile(/.*\/v(\d+(?:\.\w+)*)-(\w*)\.json$/)
+    final private Pattern VERSION_TARGZ = Pattern.compile(/.*\/pkg\/(\d+(?:\/\w+)+)\/fusion-(\w+)\.tar\.gz$/)
+
+    boolean enabled
+    String version
+
+    FusionMetadata(Session session) {
+        if( session.config.fusion as Map ) {
```

Comment:
It my be better to rely on the `FusionConfig` via [this getter](https://github.com/nextflow-io/nextflow/pull/4945/commits/4380403ef05e95b4b50691f2fe02b4272f022173) that I've added

Reply:
done